_posts/2016-04-07-developing-nectar-extensions-part-2.md

```diff
@@ -0,0 +1,187 @@
+---
+layout: post
+cover: 'assets/images/general-cover-3.jpg'
+title: Developing Nectar Extensions Part 2
+tags: docs
+subclass: 'post tag-docs'
+categories: 'elixir'
+author: 'Nimish'
+navigation: true
+logo: 'assets/images/nectar-cart.png'
+---
+
+>
+The post belongs to _NectarCommerce and Extension Framework Awareness_ Series
+>
+1. _[NectarCommerce Vision](http://vinsol.com/blog/2016/04/08/nectarcommerce-vision/)_
+1. _[Extension Framework Game Plan](http://vinsol.com/blog/2016/04/12/extension-framework-game-plan/)_
+1. _[Introduction to Metaprogramming](http://vinsol.com/blog/2016/04/14/introduction-to-metaprogramming/)_
+1. _[Ecto Model Schema Extension](http://vinsol.com/blog/2016/04/15/ecto-model-schema-extension/)_
+1. Ecto Model Support Functions Extension
+1. Phoenix Router Extension
+1. Phoenix View Extension
+1. Running Multiple Elixir Apps Together
+1. Extension Approach Explained
+1. **Developer Experience and Workflow developing Favorite Product Extension**
+1. Developer Experience and Workflow testing Favorite Product Extension
+
+Developing Nectar Extensions Part 2
+=============
+
+### Where we left off ###
+
+In our [previous approach](), we tried to compile extensions and then based on it compile a version of nectar, which had the serious limitation of Nectar was unavailable for testing.
+
+What we need is that Nectar is available and compiled while developing extensions and if an extension is added it should recompile itself to include the new extensions.
+We have modified Nectar for this approach to seek for extensions and used a custom compiler step(A story for another time) to mark files for recompilation. Let's get started and see if we can scale the testing barrier.
+
+Note: Most(read all) of the code for the extension is same. You can probably skim through it if you have gone through the previous post. Copied, pasted and modified with changes highlighted here for posterity.
+
+### A layered guide to nectar extensions ###
+
+__Setup__: Create a new phoenix application to hold the favorite products application.
+in your shell run inside the umbrella/apps folder:
+
+<script src="https://gist.github.com/nimish-mehta/994e51defad0787eb88e6611219066fb.js?file=new_phoenix_application.bash"></script>
+
+
+We could have gone with a regular mix application, but phoenix/ecto will come in handy in this case, since we want to have views to display stuff and a model to store data.
+
+While we are at it let's configure our dev.exs to use the same db as nectar, we could write some code and share the db settings between nectar and our extensions see: link to running multiple phoenix application together for more details. But now for simplicity's sake we are  just copying the settings from nectar to get started.
```

Comment:
see: link to running multiple phoenix application together for more detail

comma before see
and we should have the link here